Code/GraphMol/SmilesParse/catch_tests.cpp

```diff
@@ -2003,9 +2003,195 @@ TEST_CASE("github #5466 writing floating point atom props cxsmiles", "[smiles][c
+  }
+  SECTION("detect 2d when z coords are all zero") {
+    auto m =
+        "CC(O)Cl |(-3.9163,5.4767,0;-3.9163,3.9367,0;-2.5826,3.1667,0;-5.25,3.1667,0)|"_smiles;
+    REQUIRE(m);
+    REQUIRE(m->getNumConformers() == 1);
+    CHECK(!m->getConformer().is3D());
+    CHECK(m->getAtomWithIdx(1)->getChiralTag() ==
+          Atom::ChiralType::CHI_UNSPECIFIED);
+  }
+}
+
+TEST_CASE("wiggly and wedged bonds in CXSMILES") {
+  SECTION("basic reading/writing") {
+    auto m = "CC(O)F |w:1.2|"_smiles;
+    REQUIRE(m);
+    unsigned int bondcfg;
+    CHECK(m->getBondWithIdx(2)->getPropIfPresent("_MolFileBondCfg", bondcfg));
+    CHECK(bondcfg == 2);
+    // make sure we end up with a wiggly bond in output mol blocks:
+    reapplyMolBlockWedging(*m);
+    auto mb = MolToV3KMolBlock(*m);
+    CHECK(mb.find("CFG=2") != std::string::npos);
+    // make sure we end up with the wiggly bond in the output CXSMILES:
+    auto cxsmi = MolToCXSmiles(*m);
+    CHECK(cxsmi == "CC(O)F |w:1.2|");
+    // but we can turn that off
+    SmilesWriteParams ps;
+    cxsmi =
+        MolToCXSmiles(*m, ps, SmilesWrite::CX_ALL ^ SmilesWrite::CX_BOND_CFG);
+    CHECK(cxsmi == "CC(O)F");
+  }
+
+  SECTION("CXSMILES wiggly bond over-rides atomic stereo") {
+    auto m = "C[C@H](O)F |w:1.2|"_smiles;
+    REQUIRE(m);
+    unsigned int bondcfg;
+    CHECK(m->getBondWithIdx(2)->getPropIfPresent("_MolFileBondCfg", bondcfg));
+    CHECK(bondcfg == 2);
+    CHECK(m->getAtomWithIdx(1)->getChiralTag() ==
+          Atom::ChiralType::CHI_UNSPECIFIED);
+
+    // make sure we end up with a wiggly bond in output mol blocks:
+    reapplyMolBlockWedging(*m);
+    auto mb = MolToV3KMolBlock(*m);
+    CHECK(mb.find("CFG=2") != std::string::npos);
+    // make sure we end up with the wiggly bond in the output CXSMILES:
+    auto cxsmi = MolToCXSmiles(*m);
+    CHECK(cxsmi == "CC(O)F |w:1.2|");
+  }
+  SECTION("make sure order gets reversed when needed") {
+    auto m = "CC(O)Cl |w:1.0|"_smiles;
+    REQUIRE(m);
+    CHECK(m->getBondWithIdx(0)->getBeginAtomIdx() == 1);
+    unsigned int bondcfg;
+    CHECK(m->getBondWithIdx(0)->getPropIfPresent("_MolFileBondCfg", bondcfg));
+    CHECK(bondcfg == 2);
+  }
+
+  SECTION("make sure wedging gets applied when coordinates are there") {
+    auto m =
+        "CC(O)Cl |(-3.9163,5.4767,;-3.9163,3.9367,;-2.5826,3.1667,;-5.25,3.1667,),wU:1.0|"_smiles;
+    REQUIRE(m);
+    unsigned int bondcfg;
+    CHECK(m->getBondWithIdx(0)->getPropIfPresent("_MolFileBondCfg", bondcfg));
+    CHECK(bondcfg == 1);
+    CHECK(m->getAtomWithIdx(1)->getChiralTag() ==
+          Atom::ChiralType::CHI_TETRAHEDRAL_CW);
+    m = "CC(O)Cl |(-3.9163,5.4767,;-3.9163,3.9367,;-2.5826,3.1667,;-5.25,3.1667,),wD:1.0|"_smiles;
+    REQUIRE(m);
+    CHECK(m->getBondWithIdx(0)->getPropIfPresent("_MolFileBondCfg", bondcfg));
+    CHECK(bondcfg == 3);
+    CHECK(m->getAtomWithIdx(1)->getChiralTag() ==
+          Atom::ChiralType::CHI_TETRAHEDRAL_CCW);
+  }
```

Comment:
might be handy to have a section of bad cxsmiles to check for expected error mode. On error, this should just return nullptr, right? no exception?

    SECTION("some invalid cxsmiles") {
        auto m = "CC(O)Cl |wU:0.0,wU:1.0|"_smiles;
        CHECK(!m);
        m = "CC(O)Cl |wUP:1.0|"_smiles;
        CHECK(!m);
        m = "CC(O)Cl |wU:1|"_smiles;
        CHECK(!m);
    }

Reply:
Errors in the cxsmiles parsing do lead to exceptions. I added some tests to make sure these get thrown.